### PR TITLE
Drop libprotobuf runtime dependency

### PR DESCRIPTION
**Pull request by jtilly**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

In #6 we started linking against protobuf statically, but we kept the protobuf runtime dependency. Now that the global protobuf pinning is 3.20 but the onnxruntime only supports 3.19, the protobuf runtime dependency makes onnxruntime incompatible with many new package builds. With this PR, I'm dropping the protobuf runtime dependency, because I don't think it serves any purpose. Please correct me if I'm wrong.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -55,17 +56,16 @@ requirements:
...
     # See: https://github.com/conda-forge/onnxruntime-feedstock/issues/5
-    - libprotobuf-static
+    - libprotobuf-static 3.19.1
     - numpy
     - pybind11
     - safeint
   run:
     - coloredlogs
     - packaging
-    - protobuf
```
Comment (xhochy) [inline recipe/meta.yaml:68]: This is the python package, not the C++ one. Is that really not used?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -25,7 +26,7 @@ source:
     sha256: 5daca6ca216495edf89d167f808d1d03c4a4d929cef7da5e10f135ae1540c7e4
     folder: json
 build:
-  number: 1
+  number: 3
   skip: true  # [win]
   # Since 1.11, power9 seems to be required.
   skip: true  # [ppc64le]
```
Comment (xhochy) [inline recipe/meta.yaml:32]: You should also ignore the generated `run_exports` to make this compatible with other versions of dynamically linked protobuf.
```suggestion
  skip: true  # [ppc64le]
  ignore_run_exports_from:
    - libprotobuf
```

File: recipe/meta.yaml (hunk 1 above)
Comment (jtilly) [reply recipe/meta.yaml]: Thanks, reverted.

File: recipe/meta.yaml (hunk 2 above)
Comment (jtilly) [reply recipe/meta.yaml:31]: Thanks, done!

**Pull request merged**